chromadb/test/property/test_query_helpers.py

```diff
@@ -0,0 +1,90 @@
+from chromadb.utils.query_helper import Where, WhereDocument
+
+
+def test_where_gt() -> None:
+    where_filter = Where().gt("price", 100)
+    assert where_filter.to_filter() == {"price": {"$gt": 100}}
+
+
+def test_where_gte() -> None:
+    where_filter = Where().gte("price", 100)
+    assert where_filter.to_filter() == {"price": {"$gte": 100}}
+
+
+def test_where_lt() -> None:
+    where_filter = Where().lt("price", 100)
+    assert where_filter.to_filter() == {"price": {"$lt": 100}}
+
+
+def test_where_lte() -> None:
+    where_filter = Where().lte("price", 100)
+    assert where_filter.to_filter() == {"price": {"$lte": 100}}
+
+
+def test_where_ne() -> None:
+    where_filter = Where().ne("price", 100)
+    assert where_filter.to_filter() == {"price": {"$ne": 100}}
+
+
+def test_where_eq() -> None:
+    where_filter = Where().eq("price", 100)
+    assert where_filter.to_filter() == {"price": {"$eq": 100}}
+
+
+# def test_where_in() -> None:
+#     where_filter = Where().in_("category", ["electronics", "books"])
+#     assert where_filter.to_filter() == {"category": {"$in": ["electronics", "books"]}}
+
+
+def test_where_and() -> None:
+    where_filter = Where().and_(
+        Where().eq("category", "electronics"), Where().gt("price", 100)
+    )
+    assert where_filter.to_filter() == {
+        "$and": [{"category": {"$eq": "electronics"}}, {"price": {"$gt": 100}}]
+    }
+
+
+def test_where_or() -> None:
+    where_filter = Where().or_(
+        Where().eq("category", "electronics"), Where().eq("category", "books")
+    )
+    assert where_filter.to_filter() == {
+        "$or": [{"category": {"$eq": "electronics"}}, {"category": {"$eq": "books"}}]
+    }
+
+
+def test_where_and_or_combination() -> None:
+    where_filter = Where().and_(
+        Where().eq("category", "electronics"),
+        Where().or_(Where().gt("price", 100), Where().lt("price", 50)),
+    )
+    assert where_filter.to_filter() == {
+        "$and": [
+            {"category": {"$eq": "electronics"}},
+            {"$or": [{"price": {"$gt": 100}}, {"price": {"$lt": 50}}]},
+        ]
+    }
+
+
+def test_where_document_contains() -> None:
+    where_doc_filter = WhereDocument().contains("laptop")
+    assert where_doc_filter.to_filter() == {"$contains": "laptop"}
+
+
+def test_where_document_and() -> None:
+    where_doc_filter = WhereDocument().and_(
```

Comment:
One thing thats a bit clunky about this is that you have to call Where() for each subclause, could it be possible to do

Where().and().eq("category, "electronics").or etc?

Also I find the and construction a bit strange. Would an api that was more ordered the way you write code be possible?

Like

Where().eq("category", "electronics").and().eq("name", "john") ? Just thinking out loud on this one.